handler.go

```diff
@@ -16,11 +18,42 @@ import (
 
 // TODO: Check HTTP response codes: 400, 401, 403, 409.
 
+// getUsernameAndPassword returns the username and password encoded in
+// a request. The credentials may be present as URL query params, or as
+// a Basic Authentication header.
+func getUsernameAndPassword(r *http.Request) (string, string, error) {
+	q := r.URL.Query()
+	username, password := q.Get("u"), q.Get("p")
+	if username != "" && password != "" {
+		return username, password, nil
+	}
+	auth := r.Header.Get("Authorization")
+	if auth == "" {
+		return "", "", nil
+	}
+	fields := strings.Split(auth, " ")
+	if len(fields) != 2 {
+		return "", "", fmt.Errorf("invalid Basic Authentication header")
+	}
+	bs, err := base64.StdEncoding.DecodeString(fields[1])
+	if err != nil {
+		return "", "", fmt.Errorf("invalid Base64 encoding")
+	}
+	fields = strings.Split(string(bs), ":")
+	if len(fields) != 2 {
+		return "", "", fmt.Errorf("invalid Basic Authentication value")
+	}
+	return fields[0], fields[1], nil
+}
+
 // Handler represents an HTTP handler for the InfluxDB server.
 type Handler struct {
 	server *Server
 	mux    *pat.PatternServeMux
 
+	// Whether endpoints require authentication.
+	AuthenticationEnabled bool
```

Comment:
I don't think there's any benefit to making it private. When you make fields private then you start expanding out the constructor arg list and it gets unwieldy after a while. Personally, I try to avoid arguments that aren't required for initialization in the constructor.